src/lib/cimb_translator/Config.h

```diff
@@ -1,30 +1,105 @@
 /* This code is subject to the terms of the Mozilla Public License, v.2.0. http://mozilla.org/MPL/2.0/. */
 #pragma once
 
-// probably defaults with a pimpl to read/cache if there's a (yaml?) config locally?
+#include "GridConf.h"
+
 namespace cimbar
 {
-	namespace Config
+	class Config
 	{
-		bool dark();
-
-		unsigned color_bits();
-		unsigned symbol_bits();
-		unsigned bits_per_cell();
-		unsigned ecc_bytes();
-
-		int image_size();
-		unsigned anchor_size();
-
-		unsigned cell_size();
-		unsigned cell_spacing();
-		unsigned num_cells();
-		unsigned corner_padding();
-		unsigned interleave_blocks();
-		unsigned interleave_partitions();
-
-		unsigned fountain_chunk_size(unsigned ecc, unsigned bitspercell=0);
-		unsigned fountain_chunks_per_frame();
-		unsigned compression_level();
-	}
+	protected:
+		using GridConf = Conf8x8;
+
+	public:
+		static constexpr bool dark()
+		{
+			return true;
+		}
+
+		static constexpr unsigned color_bits()
+		{
+			return GridConf::color_bits;
+		}
+
+		static constexpr unsigned symbol_bits()
+		{
+			return GridConf::symbol_bits;
+		}
+
+		static constexpr unsigned bits_per_cell()
+		{
+			return color_bits() + symbol_bits();
+		}
+
+		static constexpr unsigned ecc_bytes()
+		{
+			return GridConf::ecc_bytes;
+		}
+
+		static constexpr unsigned ecc_block_size()
+		{
+			return GridConf::ecc_block_size;
+		}
+
+		static constexpr int image_size()
+		{
+			return GridConf::image_size;
+		}
+
+		static constexpr unsigned anchor_size()
+		{
+			return 30;
+		}
+
+		static constexpr unsigned cell_size()
+		{
+			return GridConf::cell_size;
+		}
+
+		static constexpr unsigned cell_spacing()
+		{
+			return cell_size() + 1;
+		}
+
+		static unsigned corner_padding();
+
+		static constexpr unsigned cell_offset()
+		{
+			return GridConf::cell_offset;
+		}
+
+		static constexpr unsigned cells_per_col()
+		{
+			return GridConf::cells_per_col;
+		}
+
+		static unsigned total_cells();
+
+		static unsigned decode_window_bits();
+
+		static unsigned capacity(unsigned bitspercell=0);
+
+		static constexpr unsigned interleave_blocks()
+		{
+			return ecc_block_size();
+		}
+
+		static constexpr unsigned interleave_partitions()
+		{
+			return 2;
+		}
+
+		static constexpr unsigned fountain_chunks_per_frame()
+		{
+			return 10;
```

Comment:
I think this (and interleave_partitions) needs to be in the GridConf too, and will probably do that in a followup PR.